src/llama.cpp

```diff
@@ -12692,6 +12715,15 @@ static void llama_set_inputs(llama_context & lctx, const llama_batch & batch) {
                             }
                         }
                         data[h*(n_kv*n_tokens) + j*n_kv + i] = f;
+
+                        // may need to cut off old tokens for sliding window
+                        if (data_swa && f != -INFINITY) {
```

Comment:
Won't this leave some values uninitialized?

Reply:
yup, right. Should be fixed in https://github.com/ggerganov/llama.cpp/pull/8227/commits/ab2c3de9b308b80815ba5e5b9f459f56034874e2